src/libcore/sync/atomic.rs

```diff
@@ -941,8 +942,12 @@ impl<T> AtomicPtr<T> {
     ///
     /// let other_ptr   = &mut 10;
     /// let another_ptr = &mut 10;
-    ///
-    /// let value = some_ptr.compare_and_swap(other_ptr, another_ptr, Ordering::Relaxed);
+    /// // here `some_ptr` will not change as `other_ptr` has a different address from `ptr`
+    /// some_ptr.compare_and_swap(other_ptr, another_ptr, Ordering::Relaxed);
+    /// assert_eq!( unsafe { *some_ptr.load(Ordering::Relaxed) }, 5);
+    /// // but, in this instance `some_ptr` has `ptr` address, so the value is changed.
+    /// some_ptr.compare_and_swap(ptr, another_ptr, Ordering::Relaxed);
+    /// assert_eq!(unsafe { *some_ptr.load(Ordering::Relaxed) }, 10);
```

Comment:
Please capitalize 'here' and 'but'. cc @steveklabnik I think we want comments capitalized as sentences, right?